tests/functional/test_install.py

```diff
@@ -528,7 +530,7 @@ def test_editable_install__local_dir_no_setup_py_with_pyproject(
     assert 'A "pyproject.toml" file was found' in msg
 
 
-@pytest.mark.skipif("sys.version_info >= (3,4)")
+@pytest.mark.skipif("pip._vendor.six.PY3")
```

Comment:
I would do something like

```python
python3_only = pytest mark.skipif(PY3, reason="Python 3 only")

@python3_only
def test_():
    ...
```

and get rid of the `noqa`.

Reply:
Like this? Which `noqa` are you referring to?

```diff
diff --git a/tests/functional/test_install.py b/tests/functional/test_install.py
index a8d3cf96..4d896707 100644
--- a/tests/functional/test_install.py
+++ b/tests/functional/test_install.py
@@ -530,7 +530,9 @@ def test_editable_install__local_dir_no_setup_py_with_pyproject(
     assert 'A "pyproject.toml" file was found' in msg


-@pytest.mark.skipif("pip._vendor.six.PY3")
+python3_only = pytest.mark.skipif(pip._vendor.six.PY3, reason="Python 3 only")
+
+@python3_only
 @pytest.mark.xfail
 def test_install_argparse_shadowed(script):
     # When argparse is in the stdlib, we support installing it
```

Reply:
I was referring to the noqa on the import, since now it would be seen as used. Also we can import and use `PY3` directly, I think using the fully qualified name was a workaround, right?

Reply:
Ah yes, good idea! Updated!

Reply:
Hmm, https://github.com/pypa/pip/pull/6782/commits/6d20cbca23d15f7128516617fc45db374f781ce8 gives:
```
Error evaluating 'skipif' expression
    pip._vendor.six.PY2
NameError: name 'pip' is not defined
```
https://travis-ci.org/pypa/pip/jobs/582302394#L334

Any ideas?

Reply:
I mentioned in another comment [here](https://github.com/pypa/pip/pull/6782/files#diff-e60fbd8ee5898a34b73747e99a5cf805R549), we should use a similar decorator like

```python
non_python2_only = pytest.mark.skipif(PY2, reason="Non-Python 2 only")
```

Reply:
Bingo, thanks!